pkg/pillar/controllerconn/resolver.go

```diff
@@ -361,7 +392,7 @@ func ResolveWithPortsLambda(domain string,
 					if response != nil && len(response) > 0 {
 						select {
 						case resolvedIPsChan <- response:
-						default:
```

Comment:
so we hang here indefinitely?

Reply:
No there is `case <-quit:` below which triggers when `quit` is closed, which is done in defer:
```
	defer func() {
		close(quit)
		<-wgChan
	}()
```